runtime/doc/builtin.txt

```diff
@@ -6040,6 +6042,30 @@ matchaddpos({group}, {pos} [, {priority} [, {id} [, {dict}]]])
 		Can also be used as a |method|: >
 			GetGroup()->matchaddpos([23, 11])
 
+matchall({list}, {pat} [, {submatch}])				*matchall()*
+		Returns the |List| of matches in {list} where {pat} matches.
+		{list} is a |List| of strings.  {pat} is matched against each
+		string in {list}.
+
+		Returns a |List| where each item is a |List|.  Each item in
+		the inner list has two elements:
+			The starting index of the match.
+			The matched string.
+		The inner list item is empty if there is no match.  Example: >
+			:echo matchall(['abs', 'are'], 'a.*')
+			[[[0, 'abs']], [[0, 'are']]]
```

Comment:
return a 3 level deeper list, was it difficult to use?

Reply:
Shouldn't the returned list look like:
```
[[0, 'abs'], [0, 'are']]
```

Reply:
No.  The inner list is used because you may have multiple matches within a string.  Example:
```
   :echo matchall(['a1a2', 'xa3a4'], 'a.')
   [[[0, 'a1'], [2, 'a2']], [[1, 'a3'], [3, 'a4']]]
```
We can convert the two-item list into a Dict to avoid this confusion.

Reply:
> We can convert the two-item list into a Dict to avoid this confusion.

I think, mentioning it in the help is enough (with the above example).